### PR TITLE
Revert changes that break downstream

### PR DESCRIPTION
**Pull request by timkpaine**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by timkpaine** [comment]
revert https://github.com/conda-forge/quantlib-feedstock/pull/5

**Comment by carterbox** [comment]
Sorry, this is my mistake. This is a jinja expression, so it needs `{{ }}`.

See example of similar expression in the docs. https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#referencing-subpackages

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,10 +12,10 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 3
+  number: 4
   skip: true  # [win and vc<14]
   run_exports:
-    - pin_compatible('quantlib', max_pin='x')
+    - quantlib
```
Comment (carterbox) [inline recipe/meta.yaml]:
```suggestion
    - {{ pin_compatible('quantlib', max_pin='x') }}
```

Comment (timkpaine) [reply recipe/meta.yaml]: This doesn't appear to be valid, see the quantlib python staged recipe breaking

Comment (timkpaine) [reply recipe/meta.yaml]: Ah you had forgotten to put it in {{

**Pull request merged**